File: 2294_Partition_Array_Such_That_Maximum_Difference_Is_K.py
class Solution(object):
    def partitionArray(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        nums.sort()
        num_subsequences = 0
        i = 0
        n = len(nums)
        while i < n:
            num_subsequences += 1
            current_min = nums[i]
            while i < n and nums[i] - current_min <= k:
                i += 1
        return num_subsequences

File: 2294_Partition_Array_Such_That_Maximum_Difference_Is_K.cpp
class Solution {
public:
    int partitionArray(std::vector<int>& nums, int k) {
        std::sort(nums.begin(), nums.end());
        int num_subsequences = 0;
        int i = 0;
        while (i < nums.size()) {
            num_subsequences++;
            int current_min = nums[i];
            while (i < nums.size() && nums[i] - current_min <= k) {
                i++;
            }
        }
        return num_subsequences;
    }
};
